src/iam/iam-base.ts

```diff
@@ -306,17 +306,25 @@ export class IAMBase {
     }
   }
 
-  private async getPublicKey() {
+  private async getPublicKey(): Promise<string> {
     if (this._signer) {
       const address = await this._signer.getAddress();
       const hash = keccak256(address);
+      const sig = await this._signer.signMessage(arrayify(hash));
```

Comment:
If we are not prefixing here, is there any point in hashing the address? Couldn't we just send `arrayify(address)` to the wallet? The user might then have a better idea of what they're signing. But maybe I'm overlooking something with this.

Reply:
Additionally, what do you think of `eth_signTypedData`? I think it would be a good solution to the problems we face with `eth_sign`, though I'm not sure how widespread it's supported (and indeed we would need to support it in KM).

Reply:
Haha, I thought of not hashing the address as well, but I was scared to change it in case there was a good reason. But I also can't think of a good reason and it would be nice if the user could recognize what they are hashing. @dwojno or @JGiter do you know why the address was hashed?

Reply:
Re `eth_signTypedData`, I think we only have access to the ethers (v4) Signer api, which doesn't have _signTypedData https://docs.ethers.io/v4/api-wallet.html#signer . Ethers v5 does (https://docs.ethers.io/v5/api/signer/#Signer-signTypedData) but it seems experimental so maybe a bit earlier to rely on at this point and as you say, it might not be supported widely by wallets.

Reply:
It looks like there's really no need to hash the address